graspologic/cluster/rclust.py

```diff
@@ -0,0 +1,377 @@
+            k = cluster.n_components_
+            pred = cluster.predict(X)
+
+            if self.delta_criter or self.likelihood_ratio:
+                single_cluster = AutoGMMCluster(
+                    min_components=1, max_components=1, **self.cluster_kws
+                )
+                single_cluster.fit(X)
+                criter_single_cluster = single_cluster.criter_
+                lik_single_cluster = single_cluster.model_.score(X)
+
+            if k > 1:
+                # check whether the difference between the criterion of "split"
+                # and "not split" is greater than the threshold, delta_criter
+                if self.delta_criter:
+                    if criter_single_cluster - criter < self.delta_criter:
+                        k = 1
+                # perform likelihood ratio test
+                if self.likelihood_ratio:
+                    LR = 2 * (lik - lik_single_cluster)
+                    p = chi2.sf(LR, k - 1)
+                    # TODO: maybe set a default p-value threshold if do LR test
+                    if p < self.likelihood_ratio:
+                        k = 1
+                # TODO: maybe add similar tests for Kmeans
+
+        elif self.cluster_method == "Kmeans":
+            cluster = KMeansCluster(
+                max_clusters=self.max_components, **self.cluster_kws
+            )
+            cluster.fit(X)
+            model = cluster.model_
+            k = cluster.n_clusters_
+            criter = cluster.silhouette_
+            pred = cluster.predict(X)
+
+        elif self.cluster_method == "Spherical-Kmeans":
+            # TODO: could use SphericalKMeans
+            pass
+
+        results = {"model": model, "criter": criter, "n_components": k}
+        self.results_ = results
+        self.k_ = k
+        self.model_ = model
+        self.pred_ = pred
+        self.children = []
+
+        if k > 1:
+            if self.root.labels.shape[1] < self.depth + 1:
+                self.root.labels = np.column_stack(
+                    (self.root.labels, np.zeros((len(self.root.X), 1), dtype=int))
+                )
+            self.root.labels[self.indx, self.depth] = pred
+
+            uni_labels = np.unique(pred)
+            for ul in uni_labels:
+                inds = pred == ul
+                new_x = self.X[inds]
+
+                RecursiveCluster(
+                    selection_criteria=self.selection_criteria,
+                    cluster_method=self.cluster_method,
+                    parent=self,
+                    max_components=self.max_components,
+                    min_split=self.min_split,
+                    max_level=self.max_level,
+                    cluster_kws=self.cluster_kws,
+                    delta_criter=self.delta_criter,
+                    likelihood_ratio=self.likelihood_ratio,
+                )
+                self.children[-1].X = new_x
+                self.children[-1].indx = self.indx[
+                    inds
```

Comment:
apologies for repeating a comment  I think I said something similar elsewhere) but I still don't see why either needs to be stored as an attribute, same logic I think as for the `predict` function